<launch>
    <rosparam file="$(find Naturobo2019PublicSecurityRobot_main_ros)/param/logitech_f710_joy_assign.yaml" command="load" />

    <arg name="use_joy" value="true" />
    <!-- base controller -->
    <node pkg="base_controller" type="base_controller" name="base_ctrl" output="screen">
        <param name="motor_max_acc" value="50.0" />
        <param name="motor_max_vel" value="80.0" />
        <param name="invert_x" value="false" />
        <param name="invert_y" value="false" />
        <param name="invert_z" value="false" />
        <remap from="motor0_cmd_vel" to="base/motor0_cmd_vel" />
        <remap from="motor1_cmd_vel" to="base/motor1_cmd_vel" />
        <remap from="motor2_cmd_vel" to="base/motor2_cmd_vel" />
        <remap from="motor3_cmd_vel" to="base/motor3_cmd_vel" />
    </node>
    
    <!-- joy teleop -->
    <node pkg="base_controller" type="base_teleop_joy" name="base_teleop_joy" />
    
    <!-- joy node -->
    <node respawn="true" pkg="joy" type="joy_node" name="joy_node" >
        <param name="dev" type="string" value="/dev/input/js0" />
        <param name="deadzone" value="0.12" />
    </node>
    
    <node pkg="Naturobo2019PublicSecurityRobot_main_ros" type="nr_main_semiauto" name="main" output="screen" />
    <!-- shutdown controller -->
    <node pkg="base_controller" type="shutdown_controller_joy" name="shutdown_ctrl" />
    
    <node pkg="Naturobo2019PublicSecurityRobot_main_ros" type="nr_can" name="nr_can" >
    	<remap from="motor_status" to="move_slipper/cmd" />
    </node>
    
    <!-- usb_can -->
    <node pkg="Naturobo2019PublicSecurityRobot_main_ros" type="usb_can_node_19" name="usb_can_node_19" respawn="true" output="screen">
        <param name="port" value="/dev/ttyACM0" />
    </node>
    
    
</launch>